From 87251a464857e640a9389706d12f5f16d660c454 Mon Sep 17 00:00:00 2001
From: Vitaly Zaitsev <vitaly@easycoding.org>
Date: Thu, 9 Jan 2020 15:01:30 +0100
Subject: [PATCH] Removed bundled Qt configs.

---
 qt_conf/linux.qrc | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/qt_conf/linux.qrc b/qt_conf/linux.qrc
index b9f5692..03585ec 100644
--- a/qt_conf/linux.qrc
+++ b/qt_conf/linux.qrc
@@ -1,8 +1,2 @@
 <RCC>
-  <qresource prefix="/qt/etc">
-    <file alias="qt.conf">qt_linux.conf</file>
-  </qresource>
-  <qresource prefix="/fc">
-    <file>fc-custom.conf</file>
-  </qresource>
 </RCC>
-- 
2.24.1

