diff --git a/qt_conf/fc-custom.conf b/qt_conf/fc-custom.conf
index ab5c183..6e9874b 100644
--- a/qt_conf/fc-custom.conf
+++ b/qt_conf/fc-custom.conf
@@ -6,6 +6,8 @@
 	<dir>~/.fonts</dir>
 	<dir>~/.local/share/fonts</dir>
 	<dir>/usr/X11R6/lib/X11/fonts</dir>
+	<dir>/run/host/fonts</dir>
+	<dir>/run/host/user-fonts</dir>
 	<dir prefix="xdg">fonts</dir>
 	<match target="pattern">
 		<test qual="any" name="family">
