diff --git a/cmake/ranges_options.cmake b/cmake/ranges_options.cmake
index f2a3efd2f..9bd3b6cd9 100644
--- a/cmake/ranges_options.cmake
+++ b/cmake/ranges_options.cmake
@@ -39,18 +39,18 @@ endif()
 
 CMAKE_DEPENDENT_OPTION(RANGE_V3_TESTS
   "Build the Range-v3 tests and integrate with ctest"
-  ON "${is_standalone}" OFF)
+  OFF "${is_standalone}" OFF)
 
 CMAKE_DEPENDENT_OPTION(RANGE_V3_EXAMPLES
   "Build the Range-v3 examples and integrate with ctest"
-  ON "${is_standalone}" OFF)
+  OFF "${is_standalone}" OFF)
 
 CMAKE_DEPENDENT_OPTION(RANGE_V3_PERF
   "Build the Range-v3 performance benchmarks"
-  ON "${is_standalone}" OFF)
+  OFF "${is_standalone}" OFF)
 
 CMAKE_DEPENDENT_OPTION(RANGE_V3_DOCS
   "Build the Range-v3 documentation"
-  ON "${is_standalone}" OFF)
+  OFF "${is_standalone}" OFF)
 
 mark_as_advanced(RANGE_V3_PERF)
