diff --git a/Telegram/SourceFiles/core/launcher.cpp b/Telegram/SourceFiles/core/launcher.cpp
index 0019e726e..e558465f4 100644
--- a/Telegram/SourceFiles/core/launcher.cpp
+++ b/Telegram/SourceFiles/core/launcher.cpp
@@ -208,8 +208,11 @@ void Launcher::init() {
 
 	// I don't know why path is not in QT_PLUGIN_PATH by default
 	QCoreApplication::addLibraryPath("/usr/lib/plugins");
+
 	QApplication::setApplicationName(qsl("TelegramDesktop"));
 
+	QGuiApplication::setDesktopFileName(qsl("org.telegram.desktop.desktop"));
+
 #ifndef OS_MAC_OLD
 	QApplication::setAttribute(Qt::AA_DisableHighDpiScaling, true);
 #endif // OS_MAC_OLD
