diff --git a/Telegram/SourceFiles/core/file_utilities.cpp b/Telegram/SourceFiles/core/file_utilities.cpp
index dbd46ff07..86299a9bf 100644
--- a/Telegram/SourceFiles/core/file_utilities.cpp
+++ b/Telegram/SourceFiles/core/file_utilities.cpp
@@ -151,7 +151,7 @@ void ShowInFolder(const QString &filepath) {
 
 QString DefaultDownloadPath() {
 	return QStandardPaths::writableLocation(
-		QStandardPaths::DownloadLocation)
+		QStandardPaths::TempLocation)
 		+ '/'
 		+ (Main::Session::Exists() && Auth().supportMode()
 			? "Tsupport Desktop"
diff --git a/Telegram/SourceFiles/platform/linux/specific_linux.h b/Telegram/SourceFiles/platform/linux/specific_linux.h
index 9af2a64c0..8fbedb307 100644
--- a/Telegram/SourceFiles/platform/linux/specific_linux.h
+++ b/Telegram/SourceFiles/platform/linux/specific_linux.h
@@ -11,6 +11,8 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 
 #include <signal.h>
 
+#include <QStandardPaths>
+
 namespace Data {
 class LocationPoint;
 } // namespace Data
@@ -47,7 +49,7 @@ inline constexpr bool UseMainQueueGeneric() {
 } // namespace Platform
 
 inline QString psServerPrefix() {
-	return qsl("/tmp/");
+	return QStandardPaths::writableLocation(QStandardPaths::TempLocation) + qsl("/");
 }
 inline void psCheckLocalSocket(const QString &serverName) {
 	QFile address(serverName);
