diff --git a/Telegram/gyp/Telegram.gyp b/Telegram/gyp/Telegram.gyp
index cc113bdd4..da857f1f8 100644
--- a/Telegram/gyp/Telegram.gyp
+++ b/Telegram/gyp/Telegram.gyp
@@ -87,20 +87,16 @@
     ],
 
     'defines': [
-      'AL_LIBTYPE_STATIC',
-      'AL_ALEXT_PROTOTYPES',
       'TGVOIP_USE_CXX11_LIB',
       'TDESKTOP_API_ID=<(api_id)',
       'TDESKTOP_API_HASH=<(api_hash)',
+      '__STDC_FORMAT_MACROS',
       '<!@(python -c "for s in \'<(build_defines)\'.split(\',\'): print(s)")',
     ],
 
     'include_dirs': [
       '<(src_loc)',
       '<(SHARED_INTERMEDIATE_DIR)',
-      '<(libs_loc)/breakpad/src',
-      '<(libs_loc)/lzma/C',
-      '<(libs_loc)/openal-soft/include',
       '<(libs_loc)/opus/include',
       '<(minizip_loc)',
       '<(sp_media_key_tap_loc)',
diff --git a/Telegram/gyp/helpers/common/linux.gypi b/Telegram/gyp/helpers/common/linux.gypi
index 5ec145d..62ec58a 100644
--- a/Telegram/gyp/helpers/common/linux.gypi
+++ b/Telegram/gyp/helpers/common/linux.gypi
@@ -11,7 +11,6 @@
         'linux_common_flags': [
           '-pipe',
           '-Wall',
-          '-Werror',
           '-W',
           '-fPIC',
           '-Wno-unused-variable',
@@ -85,7 +84,6 @@
       ],
       'defines': [
         '_REENTRANT',
-        'QT_STATICPLUGIN',
         'QT_PLUGIN',
       ],
       'cflags_c': [
diff --git a/Telegram/gyp/telegram/sources.txt b/Telegram/gyp/telegram/sources.txt
index 6893fc7c9..03b1737c9 100644
--- a/Telegram/gyp/telegram/sources.txt
+++ b/Telegram/gyp/telegram/sources.txt
@@ -314,8 +314,6 @@
 <(src_loc)/history/view/media/history_view_sticker.cpp
 <(src_loc)/history/view/media/history_view_theme_document.h
 <(src_loc)/history/view/media/history_view_theme_document.cpp
-<(src_loc)/history/view/media/history_view_video.h
-<(src_loc)/history/view/media/history_view_video.cpp
 <(src_loc)/history/view/media/history_view_web_page.h
 <(src_loc)/history/view/media/history_view_web_page.cpp
 <(src_loc)/history/view/history_view_compose_controls.cpp
diff --git a/Telegram/gyp/utils.gyp b/Telegram/gyp/utils.gyp
index 3669236f0..d058e67db 100644
--- a/Telegram/gyp/utils.gyp
+++ b/Telegram/gyp/utils.gyp
@@ -39,7 +39,6 @@
         ],
       }, {
         'ldflags': [
-          '-static-libstdc++',
         ],
       }],
       [ '"<(build_mac)" != "1"', {
diff --git a/Telegram/gyp/helpers/modules/qt.gypi b/Telegram/gyp/helpers/modules/qt.gypi
index e329211..7c4dd38 100644
--- a/Telegram/gyp/helpers/modules/qt.gypi
+++ b/Telegram/gyp/helpers/modules/qt.gypi
@@ -14,15 +14,11 @@
               [ 'build_osx', {
                 'qt_version%': '5.6.2',
               }, {
-                'qt_version%': '5.12.5',
+                'qt_version%': '5.13.2',
               }]
             ],
           },
-          'qt_libs_5_12_5': [
-            'qwebp',
-            'qgif',
-            'qjpeg',
-            'Qt5PrintSupport',
+          'qt_libs_5_13_2': [
             'Qt5AccessibilitySupport',
             'Qt5FontDatabaseSupport',
             'Qt5EventDispatcherSupport',
@@ -30,11 +26,9 @@
             'Qt5Network',
             'Qt5Widgets',
             'Qt5Gui',
-            'qtharfbuzz',
-            'qtlibpng',
           ],
           'qt_version%': '<(qt_version)',
-          'linux_path_qt%': '/usr/local/desktop-app/Qt-<(qt_version)',
+          'linux_path_qt%': '/usr/lib/',
         },
         'qt_version%': '<(qt_version)',
         'qt_loc_unix': '<(linux_path_qt)',
@@ -44,13 +38,9 @@
             'qt_lib_debug_postfix': 'd<(ld_lib_postfix)',
             'qt_lib_release_postfix': '<(ld_lib_postfix)',
             'qt_libs': [
-              '<@(qt_libs_5_12_5)',
+              '<@(qt_libs_5_13_2)',
               'Qt5Core',
               'Qt5WindowsUIAutomationSupport',
-              'qtmain',
-              'qwindows',
-              'qtfreetype',
-              'qtpcre2',
             ],
           }],
           [ 'build_mac', {
@@ -60,7 +50,7 @@
           }],
           [ 'build_mac and not build_osx', {
             'qt_libs': [
-              '<@(qt_libs_5_12_5)',
+              '<@(qt_libs_5_13_2)',
               'Qt5Core',
               'Qt5GraphicsSupport',
               'Qt5ClipboardSupport',
@@ -72,14 +62,11 @@
           }],
           [ 'build_osx', {
             'qt_libs': [
-              'qwebp',
-              'Qt5PrintSupport',
-              'Qt5PlatformSupport',
               'Qt5Network',
               'Qt5Widgets',
               'Qt5Gui',
               'Qt5Core',
-              'qtharfbuzzng',
+              'Qt5DBus',
               'qgenericbearer',
               'qtfreetype',
               'qtpcre',
@@ -87,32 +74,16 @@
             ],
           }],
           [ 'build_linux', {
-            'qt_lib_prefix': 'lib',
-            'qt_lib_debug_postfix': '.a',
-            'qt_lib_release_postfix': '.a',
+            'qt_lib_prefix': '',
+            'qt_lib_debug_postfix': '',
+            'qt_lib_release_postfix': '',
             'qt_libs': [
-              'qxcb',
-              'Qt5XcbQpa',
               'Qt5LinuxAccessibilitySupport',
               'Qt5ServiceSupport',
               'Qt5EdidSupport',
-              'qconnmanbearer',
-              'qgenericbearer',
-              'qnmbearer',
-              '<@(qt_libs_5_12_5)',
+              '<@(qt_libs_5_13_2)',
               'Qt5DBus',
               'Qt5Core',
-              'qtpcre2',
-              'SM',
-              'ICE',
-              'fontconfig',
-              'freetype',
-              'expat',
-              'z',
-              'xcb-shm',
-              'xcb-xfixes',
-              'xcb-render',
-              'xcb-static',
             ],
           }],
         ],
@@ -236,32 +207,17 @@
         '<(DEPTH)/helpers/platform/linux/linux_glibc_wraps.gyp:linux_glibc_wraps',
       ],
       'libraries': [
-        '-lcomposeplatforminputcontextplugin',
-        '-libusplatforminputcontextplugin',
-        '-lfcitxplatforminputcontextplugin',
-        '-lhimeplatforminputcontextplugin',
-        '-lnimfplatforminputcontextplugin',
         '<@(qt_libs_release)',
-        '<(linux_path_xkbcommon)/lib/libxkbcommon.a',
-        '<(linux_path_xkbcommon)/lib/libxkbcommon-x11.a',
         '<(PRODUCT_DIR)/obj.target/helpers/platform/linux/liblinux_glibc_wraps.a',
-        #'<(linux_lib_ssl)', # added in lib_ton
-        #'<(linux_lib_crypto)', # added in lib_ton
-        '<!@(python -c "for s in \'<(linux_lib_icu)\'.split(\' \'): print(s)")',
-        '-lxcb',
         '-lX11',
-        '-lX11-xcb',
-        '-ldbus-1',
         '-ldl',
-        '-lgthread-2.0',
         '-lglib-2.0',
         '-lpthread',
       ],
       'include_dirs': [
-        '<(qt_loc)/mkspecs/linux-g++',
+        '/usr/lib/mkspecs/linux-g++',
       ],
       'ldflags': [
-        '-static-libstdc++',
         '-pthread',
         '-rdynamic',
       ],
diff --git a/Telegram/gyp/helpers/modules/qt_moc.gypi b/Telegram/gyp/helpers/modules/qt_moc.gypi
index 33cc236..b47ff9e 100644
--- a/Telegram/gyp/helpers/modules/qt_moc.gypi
+++ b/Telegram/gyp/helpers/modules/qt_moc.gypi
@@ -15,7 +15,7 @@
       '<(SHARED_INTERMEDIATE_DIR)/<(_target_name)/moc/moc_<(RULE_INPUT_ROOT).cpp',
     ],
     'action': [
-      '<(qt_loc)/bin/moc<(exe_ext)',
+      '/usr/bin/moc',
 
       # Silence "Note: No relevant classes found. No output generated."
       '--no-notes',
diff --git a/Telegram/lib_ui/gyp/qrc_rule.gypi b/Telegram/lib_ui/gyp/qrc_rule.gypi
index eb4654c..22419d2 100644
--- a/Telegram/lib_ui/gyp/qrc_rule.gypi
+++ b/Telegram/lib_ui/gyp/qrc_rule.gypi
@@ -60,7 +60,7 @@
       '<(SHARED_INTERMEDIATE_DIR)/<(_target_name)/qrc/qrc_<(RULE_INPUT_ROOT).cpp',
     ],
     'action': [
-      '<(qt_loc)/bin/rcc<(exe_ext)',
+      '/usr/bin/rcc',
       '-name', '<(RULE_INPUT_ROOT)',
       '-no-compress',
       '<(RULE_INPUT_PATH)',
diff --git a/Telegram/gyp/telegram/linux.gypi b/Telegram/gyp/telegram/linux.gypi
index 59f6acc93..c102131c9 100644
--- a/Telegram/gyp/telegram/linux.gypi
+++ b/Telegram/gyp/telegram/linux.gypi
@@ -81,8 +81,8 @@
         ],
       }], ['not_need_gtk!="True"', {
         'cflags_cc': [
-          '<!(pkg-config 2> /dev/null --cflags gtk+-2.0)',
-          '<!(pkg-config 2> /dev/null --cflags glib-2.0)',
+          '<!(pkg-config 2> /dev/null --cflags appindicator3-0.1)',
+          '<!(pkg-config 2> /dev/null --cflags gtk+-3.0)',
         ],
       }], ['<!(pkg-config ayatana-appindicator3-0.1; echo $?) == 0', {
         'cflags_cc': [ '<!(pkg-config --cflags ayatana-appindicator3-0.1)' ],
