diff --git a/Telegram/lib_base/base/basic_types.h b/Telegram/SourceFiles/base/basic_types.h
index bd5ec4059..a6d3ce0e0 100644
--- a/Telegram/lib_base/base/basic_types.h
+++ b/Telegram/lib_base/base/basic_types.h
@@ -6,7 +6,6 @@
 //
 #pragma once

-#include "base/build_config.h"
 #include "base/ordered_set.h"
 #include "base/unique_function.h"
 #include "base/functors.h"
@@ -46,3 +45,17 @@ using float32 = float;
 using float64 = double;

 using TimeId = int32;
+
+#if defined(__GNUC__) || defined(__clang__)
+#define WARN_UNUSED_RESULT __attribute__((warn_unused_result))
+#else
+#define WARN_UNUSED_RESULT
+#endif
+
+#if defined(__GNUC__)
+#define FORCE_INLINE inline __attribute__((always_inline))
+#elif defined(_MSC_VER)
+#define FORCE_INLINE __forceinline
+#else
+#define FORCE_INLINE inline
+#endif
diff --git a/Telegram/SourceFiles/config.h b/Telegram/SourceFiles/config.h
index d3460cc08..fb4ec685d 100644
--- a/Telegram/SourceFiles/config.h
+++ b/Telegram/SourceFiles/config.h
@@ -177,7 +177,7 @@ constexpr auto ApiHash = "344583e45741c457fe1862106095a5eb";
 #endif // TDESKTOP_API_ID && TDESKTOP_API_HASH
 
 #if Q_BYTE_ORDER == Q_BIG_ENDIAN
-#error "Only little endian is supported!"
+#warning "Be aware, only little endian is supported!"
 #endif // Q_BYTE_ORDER == Q_BIG_ENDIAN
 
 #if (TDESKTOP_ALPHA_VERSION != 0)
diff --git a/Telegram/lib_ui/ui/effects/animation_value.h b/Telegram/lib_ui/ui/effects/animation_value.h
index 2d822eb33..9721da61c 100644
--- a/Telegram/lib_ui/ui/effects/animation_value.h
+++ b/Telegram/lib_ui/ui/effects/animation_value.h
@@ -97,9 +97,10 @@ TG_FORCE_INLINE int interpolate(int a, int b, float64 b_ratio) {
 	return qRound(a + float64(b - a) * b_ratio);
 }
 
-#ifdef ARCH_CPU_32_BITS
+#if !(defined _LP64 || defined _M_X64 || defined _M_ARM64 || defined _M_ALPHA)
+// _LP64 is defined by GCC, others by MSVC
 #define SHIFTED_USE_32BIT
-#endif // ARCH_CPU_32_BITS
+#endif
 
 #ifdef SHIFTED_USE_32BIT
 
