diff --git a/Telegram/SourceFiles/core/launcher.cpp b/Telegram/SourceFiles/core/launcher.cpp
index 129407523..0019e726e 100644
--- a/Telegram/SourceFiles/core/launcher.cpp
+++ b/Telegram/SourceFiles/core/launcher.cpp
@@ -206,6 +206,8 @@ void Launcher::init() {
 
 	prepareSettings();
 
+	// I don't know why path is not in QT_PLUGIN_PATH by default
+	QCoreApplication::addLibraryPath("/usr/lib/plugins");
 	QApplication::setApplicationName(qsl("TelegramDesktop"));
 
 #ifndef OS_MAC_OLD
