diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index 5882bbb..1deef46 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -30,6 +30,7 @@ if (add_hunspell_library)
 	add_checked_subdirectory(hunspell)
 endif()
 add_checked_subdirectory(iconv)
+add_checked_subdirectory(jpeg)
 if (LINUX AND NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
     add_checked_subdirectory(lxqt_qtplugin)
 endif()
diff --git a/external/jpeg/CMakeLists.txt b/external/jpeg/CMakeLists.txt
new file mode 100644
index 0000000..df81831
--- /dev/null
+++ b/external/jpeg/CMakeLists.txt
@@ -0,0 +1,13 @@
+# This file is part of Desktop App Toolkit,
+# a set of libraries for developing nice desktop applications.
+#
+# For license and copyright information please follow this link:
+# https://github.com/desktop-app/legal/blob/master/LEGAL
+
+add_library(external_jpeg INTERFACE IMPORTED GLOBAL)
+add_library(desktop-app::external_jpeg ALIAS external_jpeg)
+
+if (DESKTOP_APP_USE_PACKAGED)
+    find_package(JPEG REQUIRED)
+    target_link_libraries(external_jpeg INTERFACE JPEG::JPEG)
+endif()
diff --git a/external/webrtc/CMakeLists.txt b/external/webrtc/CMakeLists.txt
index 1273291..9cf5dbf 100644
--- a/external/webrtc/CMakeLists.txt
+++ b/external/webrtc/CMakeLists.txt
@@ -99,4 +99,5 @@ target_link_libraries(external_webrtc
 INTERFACE
     ${webrtc_libs_list}
     desktop-app::external_openssl
+    desktop-app::external_jpeg
 )
