diff --git a/CMakeLists.txt b/CMakeLists.txt
index 333c41c..cc86dcb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1657,6 +1657,7 @@ PRIVATE
     common_audio/third_party/ooura/fft_size_128/ooura_fft.cc
     common_audio/third_party/ooura/fft_size_128/ooura_fft.h
     common_audio/third_party/ooura/fft_size_128/ooura_fft_sse2.cc
+    common_audio/third_party/ooura/fft_size_128/ooura_fft_neon.cc
     common_audio/third_party/ooura/fft_size_128/ooura_fft_tables_neon_sse2.h
     common_audio/third_party/ooura/fft_size_256/fft4g.cc
     common_audio/third_party/ooura/fft_size_256/fft4g.h
@@ -2031,6 +2032,7 @@ else()
         common_audio/signal_processing/cross_correlation_neon.c
         common_audio/signal_processing/downsample_fast_neon.c
         common_audio/signal_processing/min_max_operations_neon.c
+        common_audio/third_party/ooura/fft_size_128/ooura_fft_neon.cc
         modules/audio_coding/codecs/isac/fix/source/entropy_coding_neon.c
         modules/audio_coding/codecs/isac/fix/source/filterbanks_neon.c
         modules/audio_coding/codecs/isac/fix/source/filters_neon.c
