diff --git a/Telegram/SourceFiles/config.h b/Telegram/SourceFiles/config.h
index e6eee71c..dfe3941b 100644
--- a/Telegram/SourceFiles/config.h
+++ b/Telegram/SourceFiles/config.h
@@ -237,7 +237,7 @@ static const char *ApiHash = "344583e45741c457fe1862106095a5eb";
 #endif

 #if Q_BYTE_ORDER == Q_BIG_ENDIAN
-#error "Only little endian is supported!"
+#warning "Be aware, only little endian is supported!"
 #endif // Q_BYTE_ORDER == Q_BIG_ENDIAN

 #ifndef BETA_VERSION_MACRO
diff --git a/Telegram/SourceFiles/core/basic_types.h b/Telegram/SourceFiles/core/basic_types.h
index b974ef2a..3d1e5c4f 100644
--- a/Telegram/SourceFiles/core/basic_types.h
+++ b/Telegram/SourceFiles/core/basic_types.h
@@ -25,7 +25,6 @@ Copyright (c) 2014-2017 John Preston, https://desktop.telegram.org
 #include <memory>
 #include <ctime>

-#include "base/build_config.h"
 #include "base/ordered_set.h"

 using gsl::not_null;
@@ -42,3 +41,17 @@ using float64 = double;

 #define qsl(s) QStringLiteral(s)
 #define qstr(s) QLatin1String(s, sizeof(s) - 1)
+
+#if defined(__GNUC__) || defined(__clang__)
+#define WARN_UNUSED_RESULT __attribute__((warn_unused_result))
+#else
+#define WARN_UNUSED_RESULT
+#endif
+
+#if defined(__GNUC__)
+#define FORCE_INLINE inline __attribute__((always_inline))
+#elif defined(_MSC_VER)
+#define FORCE_INLINE __forceinline
+#else
+#define FORCE_INLINE inline
+#endif
diff --git a/Telegram/SourceFiles/ui/animation.h b/Telegram/SourceFiles/ui/animation.h
index aa4f5991..88c294a4 100644
--- a/Telegram/SourceFiles/ui/animation.h
+++ b/Telegram/SourceFiles/ui/animation.h
@@ -181,9 +181,10 @@ FORCE_INLINE int interpolate(int a, int b, float64 b_ratio) {
 	return qRound(a + float64(b - a) * b_ratio);
 }

-#ifdef ARCH_CPU_32_BITS
+#if !(defined _LP64 || defined _M_X64 || defined _M_ARM64 || defined _M_ALPHA)
+// _LP64 is defined by GCC, others by MSVC
 #define SHIFTED_USE_32BIT
-#endif // ARCH_CPU_32_BITS
+#endif

 #ifdef SHIFTED_USE_32BIT

diff --git a/Telegram/gyp/telegram_sources.txt b/Telegram/gyp/telegram_sources.txt
index e90614da..a8d3547c 100644
--- a/Telegram/gyp/telegram_sources.txt
+++ b/Telegram/gyp/telegram_sources.txt
@@ -1,6 +1,5 @@
 <(src_loc)/base/algorithm.h
 <(src_loc)/base/assertion.h
-<(src_loc)/base/build_config.h
 <(src_loc)/base/flags.h
 <(src_loc)/base/enum_mask.h
 <(src_loc)/base/flat_map.h
