Submodule Telegram/ThirdParty/tgcalls contains modified content
diff --git a/Telegram/ThirdParty/tgcalls/tgcalls/group/StreamingPart.h b/Telegram/ThirdParty/tgcalls/tgcalls/group/StreamingPart.h
index 08859d6..6e0812c 100644
--- a/Telegram/ThirdParty/tgcalls/tgcalls/group/StreamingPart.h
+++ b/Telegram/ThirdParty/tgcalls/tgcalls/group/StreamingPart.h
@@ -3,6 +3,7 @@
 
 #include "absl/types/optional.h"
 #include <vector>
+#include <stdint.h>
 
 namespace tgcalls {
 
