diff --git a/meson.build b/meson.build
index f1be04e5ae..beb7f2ee17 100644
--- a/meson.build
+++ b/meson.build
@@ -3049,7 +3049,7 @@ public_programs += executable(
         'systemd-detect-virt',
         'src/detect-virt/detect-virt.c',
         include_directories : includes,
-        link_with : [libshared],
+        link_with : [libshared_static, libsystemd_static, libbasic],
         install_rpath : rootlibexecdir,
         install : true)
 
