diff --git a/Telegram/SourceFiles/payments/ui/payments_panel.cpp b/Telegram/SourceFiles/payments/ui/payments_panel.cpp
index 5e5c872de2..a27f2dcb45 100644
--- a/Telegram/SourceFiles/payments/ui/payments_panel.cpp
+++ b/Telegram/SourceFiles/payments/ui/payments_panel.cpp
@@ -12,6 +12,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "payments/ui/payments_edit_card.h"
 #include "payments/ui/payments_panel_delegate.h"
 #include "payments/ui/payments_field.h"
+#include "base/invoke_queued.h"
 #include "ui/widgets/separate_panel.h"
 #include "ui/widgets/checkbox.h"
 #include "ui/wrap/fade_wrap.h"
@@ -568,9 +569,9 @@ bool Panel::createWebview(const Webview::ThemeParams &params) {
 		return false;
 	}
 	QObject::connect(raw->widget(), &QObject::destroyed, [=] {
-		crl::on_main(this, [=] {
+		InvokeQueued(qApp, crl::guard(this, [=] {
 			showCriticalError({ "Error: WebView has crashed." });
-		});
+		}));
 	});
 
 	container->geometryValue(
diff --git a/Telegram/SourceFiles/ui/chat/attach/attach_bot_webview.cpp b/Telegram/SourceFiles/ui/chat/attach/attach_bot_webview.cpp
index 209444ff10..ab95cd024e 100644
--- a/Telegram/SourceFiles/ui/chat/attach/attach_bot_webview.cpp
+++ b/Telegram/SourceFiles/ui/chat/attach/attach_bot_webview.cpp
@@ -604,9 +604,9 @@ bool Panel::createWebview(const Webview::ThemeParams &params) {
 		return false;
 	}
 	QObject::connect(raw->widget(), &QObject::destroyed, [=] {
-		crl::on_main(this, [=] {
+		InvokeQueued(qApp, crl::guard(this, [=] {
 			showCriticalError({ "Error: WebView has crashed." });
-		});
+		}));
 	});
 
 	container->geometryValue(
